objects/src/accounts/storage/map.rs

```diff
@@ -106,41 +101,6 @@ impl StorageMap {
     pub fn insert(&mut self, key: RpoDigest, value: Word) -> Word {
         self.map.insert(key, value) // Delegate to Smt's insert method
     }
-
-    /// Applies the provided delta to this account storage.
-    ///
-    /// This method assumes that the delta has been validated by the calling method and so, no
-    /// additional validation of delta is performed.
-    pub fn apply_delta(&mut self, delta: &StorageMapDelta) -> Result<(), AccountError> {
-        // apply the updated leaves to the storage map
-        for &(key, value) in delta.updated_leaves.iter() {
-            self.set_map_item(key, value)?;
-        }
-
-        // apply the cleared leaves to the storage map
-        // currently we cannot remove leaves from the storage map, so we just set them to empty
-        for &key in delta.cleared_leaves.iter() {
-            self.set_map_item(key, EMPTY_WORD)?;
-        }
-
-        Ok(())
-    }
```

Comment:
I would probably keep the `apply_delta()` method here and just use `self.insert()` instead of `self.set_map_item()`.

Also, we should probably return the new tree root from here.

Reply:
done